### PR TITLE
refactor,feat: switch to TypeScript, add get

### PR DESCRIPTION
**Pull request by ofrobots**
* Switched the project to TypeScript.
* Added a get method.
* Added basic tests.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,16 +3,41 @@
   "version": "0.1.3",
   "description": "Support for V8's sampling heap profiler",
   "repository": "v8/sampling-heap-profiler",
-  "main": "index.js",
+  "main": "out/src/index.js",
```
Comment (JustinBeckwith) [inline package.json]: More curiosity than anything - why `out` vs `build`?

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  if (cb) {
+    promise.then(cb.bind(null, null)).catch(cb);
+  } else {
+    return promise;
+  }
+}
+
+async function writeAsync(filename: string): Promise<string> {
+  if (!profiling) {
+    return Promise.reject(new Error('profiler not running'));
```
Comment (JustinBeckwith) [inline src/index.ts]: Why Promise.reject vs just throwing inside of the async function?

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    "rootDir": ".",
+    "outDir": "out"
+  },
+  "include": [
+    "src/*.ts",
+    "src/**/*.ts",
+    "test/*.ts",
+    "test/**/*.ts"
+  ]
+}
```
Comment (JustinBeckwith) [inline tsconfig.json]: Usually this is where you "nit: newline" me :)

File: package.json (hunk 1 above)
Comment (ofrobots) [reply package.json]: `node-gyp` erases the `build/` directory on a `rebuild`. npm must do a `node-gyp rebuild` when this module gets installed. This means that any shipped javascript will get nuked when the module is installed :sweat:.

File: tsconfig.json (hunk 3 above)
Comment (ofrobots) [reply tsconfig.json]: fixed.

File: package.json (hunk 1 above)
Comment (JustinBeckwith) [reply package.json]: Should we consider opening an issue to change the default in gts?

**Pull request merged**